src/test/java/IMDbTop250Scraper.java

```diff
@@ -0,0 +1,106 @@
+import java.util.HashSet;
+import java.util.Set;
+import static com.codeborne.selenide.Condition.*;
+import static com.codeborne.selenide.Selenide.*;
+
+public class IMDbTop250Scraper {
+
+    public static void main(String[] args) {
+        try {
+            openIMDbHomePage();
+            navigateToTop250();
+            scrapeTopMovies();
+        } catch (Exception e) {
+            handleError(e);
+        }
+    }
+
+    private static void scrapeTopMovies() {
+        double totalRating = 0;
+        int validRatingsCount = 0;
+        Set<String> seenTitles = new HashSet<>();
+
+        System.out.println("Top 5 Movies:");
+        for (int i = 0; i < 5; i++) {
+            String title = getMovieTitle(i);
+            String releaseYear = getReleaseYear(i);
+            double rating = getMovieRating(i);
+
+            if (isDuplicateTitle(seenTitles, title)) {
+                System.err.println("Duplicate found: " + title);
+                continue;
+            }
+
+            seenTitles.add(title);
+
+            if (rating != -1) {
+                totalRating += rating;
+                validRatingsCount++;
+            }
+
+            printMovieDetails(i, title, releaseYear, rating);
+        }
+
+        printAverageRating(validRatingsCount, totalRating);
+    }
+
+    private static boolean isDuplicateTitle(Set<String> seenTitles, String title) {
+        return seenTitles.contains(title);
+    }
+
+    private static void printMovieDetails(int index, String title, String releaseYear, double rating) {
+        System.out.printf("%d. %s (%s) - Rating: %.1f%n", index + 1, title, releaseYear, rating);
+    }
+
+    private static void printAverageRating(int validRatingsCount, double totalRating) {
+        if (validRatingsCount > 0) {
+            System.out.printf("Average Rating: %.2f%n", totalRating / validRatingsCount);
+        } else {
+            System.out.println("No valid ratings found.");
+        }
+    }
+
+    private static String getMovieTitle(int index) {
+        return $("ul > li:nth-child(" + (index + 1) + ") h3").getText();
+    }
+
+    private static String getReleaseYear(int index) {
+        return $("ul > li:nth-child(" + (index + 1) + ") div.ipc-metadata-list-summary-item__c div div div.sc-300a8231-6.dBUjvq.cli-title-metadata > span:nth-child(1)").getText();
```

Comment:
Again.. why do you think this is a good selector?
Let's break it up, in one long line, it is something like: `ul > li:nth-child div.ipc-metadata-list-summary-item__c div div div.sc-300a8231-6.dBUjvq.cli-title-metadata > span:nth-child(1)`

What will happen, if imdb decides to add like section or whatever section in between, like
`ul > li:nth-child span span div.div.div.ipc-metadata-list-summary-item__c div div div.sc-300a8231-6.dBUjvq.cli-title-metadata > span:nth-child(1)`, that's right it will break, this is not a good practice. We should use very precise selectors to get to the element.

For me, shorter and more readable way looks like this: `.ipc-metadata-list-summary-item .cli-title-metadata-item`.get(0) from three options.